Doc/library/dbm.rst

```diff
@@ -52,7 +52,8 @@ the Oracle Berkeley DB.
    Always create a new, empty database, open for reading and writing.
 
 .. |incompat_note| replace::
-   The file formats created by :mod:`dbm.gnu` and :mod:`dbm.ndbm` are incompatible.
+   The file formats created by :mod:`dbm.gnu` and :mod:`dbm.ndbm` are incompatible
+   and can not be used interchangeably.
```

Comment:
Isn't it true for other `dbm` modules too?

Reply:
Yes they are. I guess it has to do with the sentence _"The dbm.gnu module provides an interface to the GDBM library, similar to the dbm.ndbm module"_. Because of the ndbm mention, perhaps there is need to explicitly state that the file formats are not compatible? I'm also fine with moving this note up to the general introduction. I'm also fine with keeping it here.

Reply:
If it refers to the sentence _"The dbm.gnu module provides an interface to the GDBM library, similar to the dbm.ndbm module"_, it should follow this sentence and not use the replacement text. Otherwise the sentence and the note text will dis-synchronize in future.

Reply:
I'm not worried about that. @hugovk and @ezio-melotti, what do you think?

Reply:
I don't want to stall the series of doc improvements for the dbm librarhy; Diátaxis, our guiding star regarding documentation, promotes small and frequent changes, not large and complicated rewrites. I'll address your concern in a separate PR instead, @serhiy-storchaka.

Reply:
I created #114597 to address your concerns, @serhiy-storchaka.